subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/java/JavaNonABIChangePerformanceTest.groovy

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.regression.java
+
+import org.gradle.performance.AbstractCrossVersionPerformanceTest
+import org.gradle.performance.fixture.JavaSourceFileUpdater
+import spock.lang.Unroll
+
+class JavaNonABIChangePerformanceTest extends AbstractCrossVersionPerformanceTest {
+
+    @Unroll
+    def "non-abi change in #testProject"() {
+        given:
+        runner.testProject = testProject
+        runner.tasksToRun = ['assemble']
+        runner.targetVersions = ["3.5-20170221000043+0000"]
+        runner.addBuildExperimentListener(new JavaSourceFileUpdater(1))
```

Comment:
Should replace this "source file updater" with the two updaters we have in gradle-profiler (abiChange/nonAbiChange, taking the relative path of the file to change)